addons/purchase/views/account_move_views.xml

```diff
@@ -15,7 +15,7 @@
                         invisible="state != 'draft' or move_type != 'in_invoice'"
                         readonly="state != 'draft'"
                         class="oe_edit_only"
-                        domain="partner_id and [('company_id', '=', company_id), ('partner_id.commercial_partner_id', '=', commercial_partner_id)] or [('company_id', '=', company_id)]"
+                        domain="partner_id and [('company_id', 'child_of', company_id), ('partner_id.commercial_partner_id', '=', commercial_partner_id)] or [('company_id', 'child_of', company_id)]"
```

Comment:
Would `check_company=True` on the declaration of the field allow to simplify like this?
```suggestion
                        domain="partner_id and [('partner_id.commercial_partner_id', '=', commercial_partner_id)] or []"
```

Reply:
yes indeed, thank you for this suggestion <3